PitchPerfect/RecordSounsViewController.swift

```diff
@@ -46,19 +32,46 @@ class RecordSounsViewController: UIViewController {
         try! session.setCategory(AVAudioSession.Category.playAndRecord, options: .defaultToSpeaker)
         
         try! audioRecorder = AVAudioRecorder(url: filePath!, settings: [:])
+        audioRecorder.delegate = self
         audioRecorder.isMeteringEnabled = true
         audioRecorder.prepareToRecord()
         audioRecorder.record()
     }
     
     @IBAction func stopRecording(_ sender: Any) {
-        ecordingLabel.text = "Tap to Record"
-        recordButton.isEnabled =  true
-        stopRecordingButton.isEnabled = false
+        toggleUIState(true)
+        
         audioRecorder.stop()
         let audioSession = AVAudioSession.sharedInstance()
         try! audioSession.setActive(false)
     }
     
+    func audioRecorderDidFinishRecording(_ recorder: AVAudioRecorder, successfully flag: Bool) {
+        if flag {
+            performSegue(withIdentifier: "stopRecording", sender: audioRecorder.url)
+        } else {
+            print("Finishing recording")
+        }
+    }
+    
+    func toggleUIState(_ isRecording: Bool) {
```

Comment:
Comments about first submission: here is the UI toggle abstraction in order to avoid repetition in the code